api/src/controllers/team.js

```diff
@@ -70,25 +74,47 @@ async function getTeam(req, res) {
 }
 
 /**
- * Gets the team document from the database on /team.
+ * Handle login request from /team/login
  * @param {*} req request object, containing team email and password in the body as JSON
- * @param {*} res response object, the found team document
+ * @param {*} res response object, the found teamId
  * @returns 200: the team was found
  * @returns 404: team is not found
  */
 async function loginTeam(req, res) {
-  Team.findOne({ email: req.body.email })
-    .then((team) => {
-      if (team == null) {
-        res.status(400).send('User not found');
-      } else if (team.password != req.body.password) {
-        res.status(403).send('Incorrect password');
-      } else {
-        console.log(team);
-        res.send(JSON.stringify({ team: team }));
-      }
-    })
-    .catch((err) => res.status(400).json('Error: ' + err));
+  try {
+    const foundTeam = await Team.findOne({ email: req.body.email })
+    if (!foundTeam) {
+      return res.status(400).send('Incorrect email/password'); // user not found 
+    } 
+    if (await bcrypt.compare(req.body.password, foundTeam.password)){
+      const teamObj = foundTeam.toObject(); // converts a mongoose object to a plain object 
+      // remove sensitive data 
+      delete teamObj.password 
+      const accessToken = jwt.sign(teamObj, process.env.JWT_SECRET_1 || "JWT_SECRET_1", {
+        expiresIn: '15m'
+      });
+      const refreshToken = jwt.sign(teamObj, process.env.JWT_SECRET_2 || "JWT_SECRET_2", {
+        expiresIn: '1y'
+      });
+      res.cookie('accessToken', accessToken, { 
+        httpOnly: true,
+        maxAge: 30000, // 5 mins
```

Comment:
Also since the expiry times are hardcoded 4 times, maybe it'll be easier to maintain by putting them in a file in the config folder? So we only have to change 2 values instead of 4 in that file.

Reply:
Sure, will be easier to manage